pkg/otelcollector/infra-meter/infra-meter.go

```diff
@@ -28,22 +30,48 @@ func AddInfraMeters(
 		},
 	})
 	if infraMeters == nil {
-		infraMeters = map[string]*policylangv1.InfraMeter{}
+		infraMeters = map[string]*policysyncv1.InfraMeterWrapper{}
 	}
-	for pipelineName, metricConfig := range infraMeters {
-		if err := addInfraMeter(config, pipelineName, metricConfig); err != nil {
-			return fmt.Errorf("failed to add infra metric pipeline %s: %w", pipelineName, err)
+	for prefix, infraMeterWrapper := range infraMeters {
+		for key, infraMeter := range infraMeterWrapper.GetInfraMeter() {
+			if err := addInfraMeter(
+				config, infraMeterWrapper.GetPolicyName(), infraMeterWrapper.GetTelemetryCollectorId(), prefix, key, infraMeter); err != nil {
+				return fmt.Errorf("failed to add infra metric pipeline %s: %w", fmt.Sprintf("%s/%s", prefix, key), err)
+			}
 		}
 	}
 	return nil
 }
 
 func addInfraMeter(
 	config *otelconfig.Config,
-	pipelineName string,
+	policyName string,
+	telemetryCollectorID int64,
+	telemetryCollectorName string,
+	infraMeterName string,
 	infraMeter *policylangv1.InfraMeter,
 ) error {
-	pipelineName = strings.TrimPrefix(pipelineName, "metrics/")
+	processorName := fmt.Sprintf("%s-%s-%s-%d", otelconsts.ProcessorTelemetryCollector, policyName, infraMeterName, telemetryCollectorID)
+	config.AddProcessor(processorName, map[string]any{
+		"attributes": []map[string]interface{}{
```

Comment:
Nit: you can just add this processor to inframeter.Processors & Pipeline, then you won't need to invent a name for this processor.

(Note that this would need some care to not break the "implicit pipeline" logic.

Reply:
As discussed, we will handle it in a follow up PR as we have to test the updated logic properly.